pkg/gatewayserver/upstream/ns/ns.go

```diff
@@ -0,0 +1,91 @@
+// Copyright © 2019 The Things Network Foundation, The Things Industries B.V.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package ns abstracts the V3 Network Server to the upstream.Handler interface.
+package ns
+
+import (
+	"context"
+
+	"go.thethings.network/lorawan-stack/pkg/component"
+	"go.thethings.network/lorawan-stack/pkg/errors"
+	"go.thethings.network/lorawan-stack/pkg/gatewayserver/io"
+	"go.thethings.network/lorawan-stack/pkg/ttnpb"
+	"go.thethings.network/lorawan-stack/pkg/types"
+)
+
+// Handler is the upstream handler.
+type Handler struct {
+	ctx             context.Context
+	name            string
+	c               *component.Component
+	devAddrPrefixes []types.DevAddrPrefix
+}
+
+var errNotFound = errors.DefineNotFound("network_server_not_found", "network server not found for ids `ids`")
+
+// NewHandler returns a new upstream handler.
+func NewHandler(ctx context.Context, name string, c *component.Component, devAddrPrefixes []types.DevAddrPrefix) *Handler {
```

Comment:
The `name` argument can be a host according to the documentation of the corresponding field in `pkg/gatewayserver.Config`

Reply:
My point was that it's not a name